examples/sklearnex/random_forest_regressor_spmd.py

```diff
@@ -0,0 +1,73 @@
+#===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+# sklearnex RF example for distributed systems; SPMD mode
+# run like this:
+#    mpirun -n 4 python ./random_forest_regressor_spmd.py
+
+import numpy as np
+
+import dpctl
+import dpctl.tensor as dpt
+
+from mpi4py import MPI
+from sklearnex.spmd.ensemble import RandomForestRegressor
+
+from numpy.testing import assert_allclose
+from sklearn.metrics import mean_squared_error
+
+
+def generate_X_y(par, coef_seed, data_seed):
+    ns, nf = par['ns'], par['nf']
+
+    crng = np.random.default_rng(coef_seed)
+    coef = crng.uniform(-10, 10, size=(nf,))
+
+    drng = np.random.default_rng(data_seed)
+    data = drng.uniform(-100, 100, size=(ns, nf))
+    resp = data @ coef
+
+    return data, resp, coef
+
+
+comm = MPI.COMM_WORLD
+mpi_size = comm.Get_size()
+mpi_rank = comm.Get_rank()
+
+params_train = {'ns': 1000000, 'nf': 3}
```

Comment:
Reduce number of samples

Reply:
in this case the accuracy decreases. In this case, you do not need to show ACCURACY. btw we have other examples with this number of samples https://github.com/intel/scikit-learn-intelex/blob/master/examples/sklearnex/knn_bf_classification_spmd.py#L45:L46